<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BrushDock</class>
 <widget class="QWidget" name="BrushDock">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>325</width>
    <height>370</height>
   </rect>
  </property>
  <layout class="QVBoxLayout">
   <property name="spacing">
    <number>3</number>
   </property>
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <item>
    <widget class="widgets::BrushPreview" name="preview">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>70</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>128</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="topLayout">
     <item>
      <widget class="widgets::GroupedToolButton" name="reloadButton">
       <property name="icon">
        <iconset theme="view-refresh">
         <normaloff>theme:view-refresh.svg</normaloff>theme:view-refresh.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="brushTypeButton">
       <property name="toolTip">
        <string>Brush type</string>
       </property>
       <property name="popupMode">
        <enum>QToolButton::InstantPopup</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="blendModeWrapper" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QHBoxLayout" name="blendModeLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="widgets::GroupedToolButton" name="alphaPreserve">
          <property name="toolTip">
           <string>Preserve alpha</string>
          </property>
          <property name="statusTip">
           <string>Preserve alpha</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>theme:drawpile_alpha_unlocked.svg</normaloff>
            <normalon>theme:drawpile_alpha_locked.svg</normalon>theme:drawpile_alpha_unlocked.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="blendmode">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>1</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>24</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="modeColorpick">
       <property name="toolTip">
        <string>Pick initial color from layer</string>
       </property>
       <property name="icon">
        <iconset theme="color-picker">
         <normaloff>theme:color-picker.svg</normaloff>theme:color-picker.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="modeEraser">
       <property name="toolTip">
        <string>Eraser mode</string>
       </property>
       <property name="icon">
        <iconset theme="drawpile_brusherase">
         <normaloff>theme:drawpile_brusherase.svg</normaloff>theme:drawpile_brusherase.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="paintMode">
       <property name="toolTip">
        <string>Direct drawing mode</string>
       </property>
       <property name="popupMode">
        <enum>QToolButton::InstantPopup</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QGridLayout" name="grid" columnstretch="1,0">
     <property name="horizontalSpacing">
      <number>3</number>
     </property>
     <property name="verticalSpacing">
      <number>0</number>
     </property>
     <item row="7" column="0">
      <widget class="KisSliderSpinBox" name="gainBox">
       <property name="suffix">
        <string/>
       </property>
       <property name="prefix">
        <string>Gain: </string>
       </property>
       <property name="minimum">
        <number>-180</number>
       </property>
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="KisSliderSpinBox" name="hardnessBox">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="prefix">
        <string>Hardness: </string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="widgets::GroupedToolButton" name="pressureHardness">
       <property name="toolTip">
        <string>Pressure sensitivity</string>
       </property>
       <property name="icon">
        <iconset theme="pathshape">
         <normaloff>theme:pathshape.svg</normaloff>theme:pathshape.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="KisSliderSpinBox" name="brushspacingBox">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="prefix">
        <string>Spacing: </string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>999</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="widgets::GroupedToolButton" name="pressureSize">
       <property name="toolTip">
        <string>Pressure sensitivity</string>
       </property>
       <property name="icon">
        <iconset theme="pathshape">
         <normaloff>theme:pathshape.svg</normaloff>theme:pathshape.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="KisSliderSpinBox" name="colorpickupBox">
       <property name="prefix">
        <string>Color Pickup: 1/</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>32</number>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="KisSliderSpinBox" name="opacityBox">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="prefix">
        <string>Opacity: </string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="KisSliderSpinBox" name="radiusLogarithmicBox">
       <property name="prefix">
        <string>Radius: </string>
       </property>
       <property name="maximum">
        <number>800</number>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="KisSliderSpinBox" name="brushsizeBox">
       <property name="suffix">
        <string>px</string>
       </property>
       <property name="prefix">
        <string>Size: </string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="KisSliderSpinBox" name="smudgingBox">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="prefix">
        <string>Smudging: </string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="widgets::GroupedToolButton" name="pressureSmudging">
       <property name="toolTip">
        <string>Pressure sensitivity</string>
       </property>
       <property name="icon">
        <iconset theme="pathshape">
         <normaloff>theme:pathshape.svg</normaloff>theme:pathshape.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="widgets::GroupedToolButton" name="stabilizerButton">
       <property name="icon">
        <iconset theme="application-menu">
         <normaloff>theme:application-menu.svg</normaloff>theme:application-menu.svg</iconset>
       </property>
       <property name="popupMode">
        <enum>QToolButton::InstantPopup</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="widgets::GroupedToolButton" name="pressureOpacity">
       <property name="toolTip">
        <string>Pressure sensitivity</string>
       </property>
       <property name="icon">
        <iconset theme="pathshape">
         <normaloff>theme:pathshape.svg</normaloff>theme:pathshape.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QWidget" name="stabilizationWrapper" native="true">
       <layout class="QHBoxLayout" name="stabilizerLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="KisSliderSpinBox" name="stabilizerBox">
          <property name="prefix">
           <string>Stabilizer: </string>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KisSliderSpinBox" name="smoothingBox">
          <property name="prefix">
           <string>Smoothing: </string>
          </property>
          <property name="maximum">
           <number>20</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="noPermissionLabel">
     <property name="text">
      <string notr="true">Lock reason text goes here.</string>
     </property>
     <property name="textFormat">
      <enum>Qt::RichText</enum>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="bottomSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>widgets::BrushPreview</class>
   <extends>QFrame</extends>
   <header>desktop/widgets/brushpreview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>widgets::GroupedToolButton</class>
   <extends>QToolButton</extends>
   <header>desktop/widgets/groupedtoolbutton.h</header>
  </customwidget>
  <customwidget>
   <class>KisSliderSpinBox</class>
   <extends>QSpinBox</extends>
   <header>desktop/widgets/kis_slider_spin_box.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>reloadButton</tabstop>
  <tabstop>brushTypeButton</tabstop>
  <tabstop>alphaPreserve</tabstop>
  <tabstop>blendmode</tabstop>
  <tabstop>modeColorpick</tabstop>
  <tabstop>modeEraser</tabstop>
  <tabstop>paintMode</tabstop>
  <tabstop>brushsizeBox</tabstop>
  <tabstop>pressureSize</tabstop>
  <tabstop>radiusLogarithmicBox</tabstop>
  <tabstop>opacityBox</tabstop>
  <tabstop>pressureOpacity</tabstop>
  <tabstop>hardnessBox</tabstop>
  <tabstop>pressureHardness</tabstop>
  <tabstop>smudgingBox</tabstop>
  <tabstop>pressureSmudging</tabstop>
  <tabstop>colorpickupBox</tabstop>
  <tabstop>brushspacingBox</tabstop>
  <tabstop>gainBox</tabstop>
  <tabstop>stabilizerBox</tabstop>
  <tabstop>smoothingBox</tabstop>
  <tabstop>stabilizerButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
